File: backend/src/main/java/me/performancereservation/domain/admin/handler/AdminAuthSuccessHandler.java
package me.performancereservation.domain.admin.handler;

import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.security.core.Authentication;
import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
import org.springframework.stereotype.Component;

import java.io.IOException;

@Component
@RequiredArgsConstructor
@Slf4j
public class AdminAuthSuccessHandler implements AuthenticationSuccessHandler {

    //로그인이 성공한 경우 OK를 반환하는 핸들러
    @Override
    public void onAuthenticationSuccess(HttpServletRequest request,
                                        HttpServletResponse response,
                                        Authentication authentication) throws IOException, ServletException {
        //로그인이 성공하면 OK 반환
        response.setStatus(HttpStatus.OK.value());
    }
}

File: backend/src/main/kotlin/me/performancereservation/global/security/admin/handler/AdminAuthSuccessHandler.kt
package me.performancereservation.global.security.admin.handler

import io.github.oshai.kotlinlogging.KotlinLogging
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.http.HttpStatus
import org.springframework.security.core.Authentication
import org.springframework.security.web.authentication.AuthenticationSuccessHandler
import org.springframework.stereotype.Component

@Component
class AdminAuthSuccessHandler : AuthenticationSuccessHandler {

    // 로그인이 성공한 경우 OK를 반환하는 핸들러
    override fun onAuthenticationSuccess(
        request: HttpServletRequest,
        response: HttpServletResponse,
        authentication: Authentication
    ) {
        // 로그인이 성공하면 OK 반환
        response.status = HttpStatus.OK.value()
    }
}